### PR TITLE
Create text render component

### PR DESCRIPTION
**Pull request by Bernatmago**
- Created text rendering using a modified version of GLFont (recommended in slides)
- Small fix to loading transform2d and debug mode input when editing
- Added glm dependency bcs it was used by GLFont i will try replacing all glm operations with Mathgeolib to get rid of it since the library is small

### CODE REVIEW

File: Source/src/components/ComponentText.cpp [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        label->setSize(size.x, size.y);
+        label->setPosition(pos.x + (windowWidth / 2), pos.y + (windowHeight / 2));
+        dirty = false;
+    }
+}
+
+void Hachiko::ComponentText::BuildLabel(ComponentTransform2D* transform)
+{
+    // TODO: Get From transform
+    int startX = 100;
```
Comment (Erick9Thor) [inline Source/src/components/ComponentText.cpp:115]: Use the transform from componentTransform 2d hehe

File: Source/src/components/ComponentText.h [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        std::string label_text = "Sample Text";
+
+        float4 font_color = float4::one;
+        float font_size = 28.f;
+
+        char font_filename_buffer[MAX_PATH] = "Image Filename\0";
+    };
+} // namespace Hachiko
+
+inline Hachiko::Component::Type Hachiko::ComponentText::GetType()
```
Comment (Erick9Thor) [inline Source/src/components/ComponentText.h:55]: image

Why not use this?

File: Source/src/modules/ModuleTexture.h [hunk 3]
```diff
@@ -55,5 +69,7 @@ namespace Hachiko
         const short devil_version = IL_VERSION;
         static unsigned int LoadImg(const char* path, bool flip = true);
         static void DeleteImg(unsigned& img_id);
+
+        FT_Library freetype_lib;
```
Comment (Erick9Thor) [inline Source/src/modules/ModuleTexture.h:73]: Should be a pointer?

**Review by Erick9Thor — APPROVED** [review]

File: Source/src/modules/ModuleTexture.h (hunk 3 above)
Comment (Bernatmago) [reply Source/src/modules/ModuleTexture.h:73]: Nope, it seems to internally manage the instance

File: Source/src/modules/ModuleDebugMode.cpp [hunk 4]
```diff
@@ -63,51 +63,59 @@ UpdateStatus Hachiko::ModuleDebugMode::Update(const float delta)
         if (line.Intersects(plane, &d))
         {
             player = FindPlayer();
-            player->GetTransform()->SetGlobalPosition(line.GetPoint(d));
+            if (player)
+				player->GetTransform()->SetGlobalPosition(line.GetPoint(d));
```
Comment (baransrc) [inline Source/src/modules/ModuleDebugMode.cpp:67]:
```suggestion
            if (player)
            {
                player->GetTransform()->SetGlobalPosition(line.GetPoint(d));
            }
```

File: Source/src/modules/ModuleProgram.cpp [hunk 5]
```diff
@@ -108,26 +109,32 @@ Hachiko::Program* Hachiko::ModuleProgram::CreateProgram(const char* vtx_shader_p
 
 Hachiko::Program* Hachiko::ModuleProgram::CreateMainProgram()
 {
-    main_program = CreateProgram(ASSETS_FOLDER "/Shaders/vertex.glsl", ASSETS_FOLDER "/Shaders/fragment.glsl");
+    main_program = CreateProgram(ASSETS_FOLDER "/shaders/vertex.glsl", ASSETS_FOLDER "/shaders/fragment.glsl");
```
Comment (baransrc) [inline Source/src/modules/ModuleProgram.cpp:112]: Thank you so much for this <3

**Review by baransrc — CHANGES_REQUESTED** [review]
Thanks for your work, we are good to go after resolving the requested changes 💯

**Review by baransrc — APPROVED** [review]
Thanks for the hard work!!!

**Pull request merged**